src/main/java/org/spongepowered/common/util/SpongeColor.java

```diff
@@ -0,0 +1,186 @@
+    private final int green;
+    private final int blue;
+
+    private SpongeColor(int red, int green, int blue) {
+        this.rgb = red << 16 | green << 8 | blue;
+        this.red = red;
+        this.green = green;
+        this.blue = blue;
+    }
+
+    @Override
+    public int getRgb() {
+        return this.rgb;
+    }
+
+    @Override
+    public int getRed() {
+        return this.red;
+    }
+
+    @Override
+    public int getGreen() {
+        return this.green;
+    }
+
+    @Override
+    public int getBlue() {
+        return this.blue;
+    }
+
+    @Override
+    public int getContentVersion() {
+        return 1;
+    }
+
+    @Override
+    public DataContainer toContainer() {
+        return DataContainer.createNew()
+                .set(Queries.CONTENT_VERSION, this.getContentVersion())
+                .set(Queries.COLOR_RED, this.getRed())
+                .set(Queries.COLOR_GREEN, this.getGreen())
+                .set(Queries.COLOR_BLUE, this.getBlue());
+    }
+
+    @Override
+    public int hashCode() {
+        return rgb;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        return this == obj || obj != null && getClass() == obj.getClass() && hashCode() == obj.hashCode();
+    }
+
+    @Override
+    public String toString() {
+        return com.google.common.base.MoreObjects.toStringHelper(this)
+                .add("rgb", this.rgb)
+                .add("red", this.red)
+                .add("green", this.green)
+                .add("blue", this.blue)
+                .toString();
+    }
+
+    public static final class Builder implements Color.Builder {
+
+        private int red;
+        private int green;
+        private int blue;
+
+        @Override
+        public Builder rgb(int rgb) {
+            return rgb(rgb >> 16, rgb >> 8, rgb);
+        }
+
+        @Override
+        public Builder rgb(int red, int green, int blue) {
+            this.red = red & 0xFF;
+            this.green = green & 0xFF;
+            this.blue = blue & 0xFF;
+            return this;
+        }
+
+        @Override
+        public Builder mix(Color... colors) {
+            Validate.noNullElements(colors, "No null colors allowed!");
+            checkArgument(colors.length > 0, "Cannot have an empty color array!");
+            if (colors.length == 1) {
+                return from(colors[0]);
+            }
+            reset();
+            for (Color color : colors) {
+                red += color.getRed();
+                green += color.getGreen();
+                blue += color.getBlue();
+            }
+            return rgb(Math.round((float) red / colors.length), Math.round((float) green / colors.length), Math.round((float) blue / colors.length));
+        }
+
+        @Override
+        public Builder from(Color value) {
+            return rgb(red, green, blue);
+        }
+
+        @Override
+        public Builder reset() {
+            return rgb(0, 0, 0);
```

Comment:
A little wasteful when all you need to do is just set the fields and return this...

Reply:
That's true, just seemed simpler to me to do it like that. It might also be advantageous to us to use `-1` as the default value, that way we know a color has been set.